lib/p2p/Pool.ts

```diff
@@ -411,18 +413,50 @@ class Pool extends EventEmitter {
     if (!deal) {
       return;
     }
-    if (!deal.r_preimage) {
-      // TODO: Handle this unexpected case, if we are the maker we should have the preimage.
-      return;
-    }
-    const body: packets.SwapResponsePacketBody = {
-      r_preimage: deal.r_preimage,
-    };
-    const swapResponsePacket = new packets.SwapResponse(body, requestPacket.header.id);
 
-    this.logger.debug('sending back to peer: '  + JSON.stringify(body));
+    let cmdLnd = this.lndLtcClient;
+
+  	switch (deal.makerCurrency) {
+    case 'BTC':
+      break;
+    case 'LTC':
+      cmdLnd = this.lndBtcClient;
+      break;
+  	}
+  	if (!cmdLnd) {
+  	  return;
+  	}
+
+    const request = new lndrpc.SendRequest();
+  	request.setAmt(deal.takerAmount);
+  	request.setDestString(deal.takerPubKey);
+  	request.setPaymentHashString(String(deal.r_hash));
+
+    // TODO: use timeout on call
+
+    try {
+      const response = await cmdLnd.sendPaymentSync(request);
+      if (response.getPaymentError()) {
+        this.logger.error('Got error from sendPaymentSync: ' + response.getPaymentError() + ' ' + JSON.stringify(request.toObject()));
+        return;
+      }
+
+      if (!response.getPaymentPreimage()) {
+        // TODO: Handle this unexpected case, if we are the maker we should have the preimage.
```

Comment:
I think as long as we didn't have a payment error, we should always get back a payment preimage from lnd (otherwise something went very wrong with lnd) so I don't know if we need this check here.

Edit: If we do want to keep this check, change the todo above because this error doesn't have to do with being maker, it would be due to lnd not giving us a preimage on a successful payment (which again should never happen).

Reply:
I removed it. Don't think it is needed.